velox/functions/prestosql/ArrayPosition.cpp

```diff
@@ -137,8 +138,9 @@ void applyTypedFirstMatch(
 
     int i;
     for (i = 0; i < size; i++) {
-      if (!elementsBase->isNullAt(offset + i) &&
-          elementsBase->equalValueAt(searchBase, offset + i, searchIndex)) {
+      if (!elementsDecoded.isNullAt(offset + i) &&
+          elementsBase->equalValueAt(
```

Comment:
nit: it would be also nice to define equalValueAt for decoded functions

Reply:
OK, got it. Should we add it in a separate PR or in this one?

Reply:
Add a wrapper method `DecodeVector::equalValueAt`, which uses `BaseVector::equalValueAt` and `DecodeVector::indices` to compare their base vectors. Is this solution appropriate?

Reply:
CC: @bikramSingh91

Reply:
> @laithsakka @duanmeng Would it make sense to keep this PR simple and consider new APIs on DecodedVector separately?

@mbasmanova OK got it. I will add the new API on a separate pr.  cc @laithsakka

Reply:
@laithsakka @duanmeng Just FYI, we can also use DecodeVector::index(offset + i) instead of generating elementIndices since we will end up generating indices for identity and constant encodings which is redundant overhead. Alternatively, we can decide to handle these cases separately (have an if condition for each and only use elementIndices for dictionary vectors). Adding and using an API like DecodeVector::equalValueAt() will equate to the first option so looking forward to that follow up change. @duanmeng can you please make sure to update these callsites when you add the API?

Reply:
@bikramSingh91 OK got it.